src/sqlfluff/rules/L026.py

```diff
@@ -76,3 +84,14 @@ def _lint_references_and_aliases(
                     )
                 )
         return violation_buff or None
+
+    def _eval(self, segment, parent_stack, dialect, **kwargs):
+        """Override Rule L020 for dialects that use structs.
+
+        Some dialects use structs (e.g. column.field) which look like
+        table references and so incorrectly trigger this rule.
+        """
+        if dialect.name in ["bigquery"] and not self.force_enable:
```

Comment:
Should we consider building in this disabling behavior at the core level to minimize having to add rule-specific code? This code is reasonable, but I'm thinking ahead if we end up doing this for several more rules.

E.g. could rules disable themselves through class-level config or code, e.g.:
```
 _default_disabled = {"dialects": ["bigquery"]}
```

This would be similar in spirit to `_works_on_unparseable`, for example:
https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/rules/L018.py#L39

Reply:
Potentially, but then would need to add `force_enable` config to all rules if we wanted to do that.

The other option is to have config at a higher level than rules:

e.g.

```yaml
dialect-exclusions:
   bigquery: L026, L028
   all: L033

dialect-inclusions:
   ansi: L033
   bigquery: L033
   hive: L033
```

Anyway needs some thought, so let's not do that now, but certainly one to think about as we add more dialect-specific rules.

Another thought I have is to avoid the whole "rules are on by default" mentality we have now. I think some should be more advanced/niche and be disabled by default with option to enable.

Anyway, bigger discussions for another time. Will leave this as is for now if you don't mind.